src/SAML2/XML/saml/AuthnContext.php

```diff
@@ -0,0 +1,266 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SAML2\XML\saml;
+
+use DOMElement;
+use SAML2\Constants;
+use SAML2\DOMDocumentFactory;
+use SAML2\Utils;
+use SAML2\XML\saml\AuthenticatingAuthority;
+use SAML2\XML\saml\AuthnContextClassRef;
+use SAML2\XML\saml\AuthnContextDeclRef;
+use SAML2\XML\saml\AuthnContextRef;
+use Webmozart\Assert\Assert;
+
+/**
+ * Class representing SAML2 AuthnContext
+ *
+ * @author Tim van Dijen, <tvdijen@gmail.com>
+ * @package SimpleSAMLphp
+ */
+class AuthnContext extends \SAML2\XML\AbstractConvertable
+{
+    /** @var \SAML2\XML\saml\AuthnContextClassRef|null */
+    private $authnContextClassRef = null;
+
+    /** @var \SAML2\XML\saml\AuthnContextDeclRef|null */
+    private $authnContextDeclRef = null;
+
+    /** @var \SAML2\XML\saml\AuthnContextDecl|null */
+    private $authnContextDecl = null;
+
+    /** @var \SAML2\XML\saml\AuthenticatingAuthority[]|null */
+    private $authenticatingAuthorities = null;
+
+
+    /**
+     * Initialize an AuthnContext.
+     *
+     * @param \SAML2\XML\saml\AuthnContextClassRef|null $authnContextClassRef
+     * @param \SAML2\XML\saml\AuthnContextDecl|null $authnContextDecl
+     * @param \SAML2\XML\saml\AuthnContextDeclRef|null $authnContextDeclRef
+     * @param \SAML2\XML\saml\AuthenticatingAuthority[]|null $authenticatingAuthorities
+     */
+    public function __construct(
+        ?AuthnContextClassRef $authnContextClassRef,
+        ?AuthnContextDecl $authnContextDecl,
+        ?AuthnContextDeclRef $authnContextDeclRef,
+        ?array $authenticatingAuthorities
+    ) {
+        if (!is_null($authnContextClassRef)) {
+            Assert::oneOf(
```

Comment:
Does `oneOf` allow the case of both being `null`? Because in this case, the minimum occurs is set to zero, so both could be skipped in fact.

Reply:
Yes, that is allowed. I've double checked by adding an `Assert::oneOf(null, [null, null], 'This is wrong');` and it didn't trigger the error message..

Reply:
Perfect!